File: blockchain_flutter/lib/marketplace_abi.dart
const String marketplaceAbi = '''[
  {
    "inputs": [{"internalType": "string", "name": "metadataURI", "type": "string"}, {"internalType": "uint256", "name": "price", "type": "uint256"}],
    "name": "createProduct",
    "outputs": [{"internalType": "uint256", "name": "productId", "type": "uint256"}],
    "stateMutability": "nonpayable",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "uint256", "name": "productId", "type": "uint256"}],
    "name": "buyProduct",
    "outputs": [{"internalType": "uint256", "name": "orderId", "type": "uint256"}],
    "stateMutability": "payable",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "uint256", "name": "orderId", "type": "uint256"}],
    "name": "markShipped",
    "outputs": [],
    "stateMutability": "nonpayable",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "uint256", "name": "orderId", "type": "uint256"}],
    "name": "completeOrder",
    "outputs": [],
    "stateMutability": "nonpayable",
    "type": "function"
  },
  {
    "inputs": [],
    "name": "withdraw",
    "outputs": [],
    "stateMutability": "nonpayable",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "uint256", "name": "productId", "type": "uint256"}],
    "name": "getProduct",
    "outputs": [{"components": [{"internalType": "uint256", "name": "id", "type": "uint256"}, {"internalType": "address", "name": "seller", "type": "address"}, {"internalType": "string", "name": "metadataURI", "type": "string"}, {"internalType": "uint256", "name": "price", "type": "uint256"}, {"internalType": "bool", "name": "active", "type": "bool"}], "internalType": "struct Marketplace.Product", "name": "product", "type": "tuple"}],
    "stateMutability": "view",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "uint256", "name": "orderId", "type": "uint256"}],
    "name": "getOrder",
    "outputs": [{"components": [{"internalType": "uint256", "name": "id", "type": "uint256"}, {"internalType": "uint256", "name": "productId", "type": "uint256"}, {"internalType": "address", "name": "buyer", "type": "address"}, {"internalType": "address", "name": "seller", "type": "address"}, {"internalType": "uint256", "name": "amount", "type": "uint256"}, {"internalType": "uint8", "name": "status", "type": "uint8"}], "internalType": "struct Marketplace.Order", "name": "order", "type": "tuple"}],
    "stateMutability": "view",
    "type": "function"
  },
  {
    "inputs": [],
    "name": "nextProductId",
    "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
    "stateMutability": "view",
    "type": "function"
  },
  {
    "inputs": [{"internalType": "address", "name": "seller", "type": "address"}],
    "name": "withdrawableBalance",
    "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
    "stateMutability": "view",
    "type": "function"
  }
]''';

File: api/src/marketplaceAbi.ts
export const marketplaceAbi = [
  {
    type: 'function',
    name: 'createProduct',
    inputs: [
      { name: 'metadataURI', type: 'string' },
      { name: 'price', type: 'uint256' }
    ],
    outputs: [{ name: 'productId', type: 'uint256' }],
    stateMutability: 'nonpayable'
  },
  {
    type: 'function',
    name: 'getOrder',
    inputs: [{ name: 'orderId', type: 'uint256' }],
    outputs: [
      {
        name: 'order',
        type: 'tuple',
        components: [
          { name: 'id', type: 'uint256' },
          { name: 'productId', type: 'uint256' },
          { name: 'buyer', type: 'address' },
          { name: 'seller', type: 'address' },
          { name: 'amount', type: 'uint256' },
          { name: 'status', type: 'uint8' }
        ]
      }
    ],
    stateMutability: 'view'
  },
  {
    type: 'function',
    name: 'nextProductId',
    inputs: [],
    outputs: [{ name: '', type: 'uint256' }],
    stateMutability: 'view'
  },
  {
    type: 'event',
    name: 'ProductCreated',
    inputs: [
      { name: 'productId', type: 'uint256', indexed: true },
      { name: 'seller', type: 'address', indexed: true },
      { name: 'metadataURI', type: 'string', indexed: false },
      { name: 'price', type: 'uint256', indexed: false }
    ]
  },
  {
    type: 'event',
    name: 'ProductPurchased',
    inputs: [
      { name: 'orderId', type: 'uint256', indexed: true },
      { name: 'productId', type: 'uint256', indexed: true },
      { name: 'buyer', type: 'address', indexed: true },
      { name: 'amount', type: 'uint256', indexed: false }
    ]
  },
  {
    type: 'event',
    name: 'OrderShipped',
    inputs: [{ name: 'orderId', type: 'uint256', indexed: true }]
  },
  {
    type: 'event',
    name: 'OrderCompleted',
    inputs: [{ name: 'orderId', type: 'uint256', indexed: true }]
  },
  {
    type: 'event',
    name: 'SellerWithdrawal',
    inputs: [
      { name: 'seller', type: 'address', indexed: true },
      { name: 'amount', type: 'uint256', indexed: false }
    ]
  }
] as const;
